### ISSUE TITLE
How can I prevent the terminal from automatically gaining focus when running tests with the C# Dev Kit?

### ISSUE THREAD

**Issue by YutaroYamada** [microsoft/vscode-dotnettools#1695]
I am using VSCode with the C# Dev Kit and running tests from the Test Explorer. 
Each time I run a test, dotnet build is triggered, causing the terminal in the panel to become active. 

However, I want to keep the test results panel visible instead because I need to quickly review the standard output of each test. 

Is there a VSCode setting to prevent this behavior?

**Comment by rwestgeest** [comment]
I have the same question. Popping up a terminal with no useful information is annoying when doing TDD. I cannot seem to find the option to prevent that.

**Comment by lbrecheisen** [comment]
Agreed. Would be nice if the build task was updated to never reveal.

**Comment by ocallesp** [comment]
This is currently is not available.

This would be a enhancement.

**Comment by bkacena** [comment]
I would like this as a feature as well.

**Comment by mpartipilo** [comment]
+1

The only half-decent workaround I have found for this is to move the terminal to the Editor area.

**Comment by florestankorp** [comment]
This behavior is very annoying because I lose focus of the task at hand, which is testing. Like @bkacena said:

> [...] only opening the terminal if the build fails [...]

...would be a great idea.

**Comment by florestankorp** [comment]
I've found a few similar issues:

https://github.com/microsoft/vscode-dotnettools/issues/198
https://github.com/microsoft/vscode-dotnettools/issues/807
https://github.com/microsoft/vscode-dotnettools/issues/1695
https://github.com/microsoft/vscode-dotnettools/issues/2183

I'd say let's join forces and focus on one, so we can get it prioritized on to the backlog.

PS: I also sent @timheuer a tweet (?) on BlueSky :)

**Comment by AbhitejJohn** [comment]
Sorry, this took us a bit to get to but this should now be fixed on the latest insider 1.73.14 and above.

**Comment by florestankorp** [comment]
Hi @AbhitejJohn, thanks for the update but the current UX is still a little confusing because the build in the terminal pops up for a second before switching to test results. It also forces the sidebar open and always shows "Test Results", rather than staying in the current view, which I would prefer to be honest.

**Comment by AbhitejJohn** [comment]
@florestankorp : Apologies for the delay. VS Code does not really give us a lot of control over that unfortunately. There is a way of disabling certain windows from popping up though that might be helpful. Here is the setting for the test results window:
```
{
  "testing.automaticallyOpenTestResults": "neverOpen"
} 
```
Unfortunately, nothing for the build task yet. That could be something we can look into providing.

**Comment by Xan-Kun** [comment]
Still very disruptive and the main reason I do not use this test runner UI. 2 years later, and nothing changed. Why was this special way of doing tests taken in the first place? why not just use the built-in task definitions?
